packages/video_player/video_player_android/example/integration_test/video_player_test.dart

```diff
@@ -38,132 +30,121 @@ String getUrlForAssetAsNetworkSource(String assetKey) {
 
 void main() {
   IntegrationTestWidgetsFlutterBinding.ensureInitialized();
+  late AndroidVideoPlayer player;
 
-  late MiniController controller;
-  tearDown(() async => controller.dispose());
-
-  group('asset videos', () {
-    setUp(() {
-      controller = MiniController.asset(_videoAssetKey);
-    });
-
-    testWidgets('registers expected implementation',
-        (WidgetTester tester) async {
-      AndroidVideoPlayer.registerWith();
-      expect(VideoPlayerPlatform.instance, isA<AndroidVideoPlayer>());
-    });
+  setUp(() async {
+    player = AndroidVideoPlayer();
+    await player.init();
+  });
 
-    testWidgets('can be initialized', (WidgetTester tester) async {
-      await controller.initialize();
+  testWidgets('registers expected implementation', (_) async {
+    AndroidVideoPlayer.registerWith();
+    expect(VideoPlayerPlatform.instance, isA<AndroidVideoPlayer>());
+  });
 
-      expect(controller.value.isInitialized, true);
-      expect(await controller.position, Duration.zero);
-      expect(controller.value.duration,
-          const Duration(seconds: 7, milliseconds: 540));
-    });
+  testWidgets('initializes at the start', (_) async {
+    final int textureId = (await player.create(DataSource(
+      sourceType: DataSourceType.asset,
+      asset: _videoAssetKey,
+    )))!;
 
-    testWidgets('can be played', (WidgetTester tester) async {
-      await controller.initialize();
+    expect(
+      await _getDuration(player, textureId),
+      const Duration(seconds: 7, milliseconds: 540),
+    );
 
-      await controller.play();
-      await tester.pumpAndSettle(_playDuration);
+    await player.dispose(textureId);
+  });
 
-      expect(await controller.position, greaterThan(Duration.zero));
-    });
+  testWidgets('can be played', (WidgetTester tester) async {
+    final int textureId = (await player.create(DataSource(
+      sourceType: DataSourceType.asset,
+      asset: _videoAssetKey,
+    )))!;
 
-    testWidgets('can seek', (WidgetTester tester) async {
-      await controller.initialize();
+    await player.play(textureId);
+    await tester.pumpAndSettle(_playDuration);
 
-      await controller.seekTo(const Duration(seconds: 3));
+    expect(await player.getPosition(textureId), greaterThan(Duration.zero));
+    await player.dispose(textureId);
+  });
 
-      expect(await controller.position, const Duration(seconds: 3));
-    });
+  testWidgets('can seek', (_) async {
+    final int textureId = (await player.create(DataSource(
+      sourceType: DataSourceType.asset,
+      asset: _videoAssetKey,
+    )))!;
 
-    testWidgets('can be paused', (WidgetTester tester) async {
```

Comment:
Should we add back a test for pausing like this one that was removed?

Reply:
Looks like there was an additional call to `await tester.pumpAndSettle(_playDuration)` after pause was called in the original test. Maybe that explains the slight difference in `player.getPosition(textureId)` and  `pausedDuration` causing the test failure?

Reply:
Hmm, let's try it :)